patterns/kubecost/main.tf

```
@@ -0,0 +1,410 @@
+  force_destroy = true
+
+  tags = local.tags
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "athena_results" {
+  bucket = aws_s3_bucket.athena_results.id
+  rule {
+    id = "cost"
+    expiration {
+      days = 1
+    }
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_ownership_controls" "athena_results" {
+  bucket = aws_s3_bucket.athena_results.id
+
+  rule {
+    object_ownership = "BucketOwnerPreferred"
+  }
+}
+
+resource "aws_s3_bucket_acl" "athena_results" {
+  depends_on = [aws_s3_bucket_ownership_controls.athena_results]
+
+  bucket = aws_s3_bucket.athena_results.id
+  acl    = "private"
+}
+
+################################################################################
+# Kubecost
+################################################################################
+
+module "kubecost_irsa" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.30"
+
+  role_name_prefix           = "${local.name}-"
+  assume_role_condition_test = "StringLike"
+
+  role_policy_arns = {
+    policy = aws_iam_policy.kubecost.arn
+  }
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kubecost:*"]
+    }
+  }
+
+  tags = local.tags
+}
+
+resource "aws_iam_policy" "kubecost" {
+  name   = local.name
+  policy = data.aws_iam_policy_document.combined.json
+}
+
+data "aws_iam_policy_document" "combined" {
+  source_policy_documents = [
+    data.aws_iam_policy_document.kubecost.json,
+    data.aws_iam_policy_document.athena.json
+  ]
+}
+
+data "aws_iam_policy_document" "kubecost" {
+  statement {
+    sid    = "KubecostSavingsAccess"
+    effect = "Allow"
+    actions = [
+      "ec2:DescribeAddresses",
+      "ec2:DescribeVolumes"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:ListAllMyBuckets",
+    ]
+    resources = ["*"]
+  }
+}
+
+data "aws_iam_policy_document" "athena" {
+  statement {
+    sid       = "athenaaccess"
+    effect    = "Allow"
+    actions   = ["athena:*"]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "ReadAccessToAthenaCurDataViaGlue"
+    effect = "Allow"
+    actions = [
+      "glue:GetDatabase*",
+      "glue:GetTable*",
+      "glue:GetPartition*",
+      "glue:GetUserDefinedFunction",
+      "glue:BatchGetPartition"
+    ]
+    resources = [
+      "arn:aws:glue:*:*:catalog",
+      "arn:aws:glue:*:*:database/athenacurcfn*",
+      "arn:aws:glue:*:*:table/athenacurcfn*/*"
+    ]
+  }
+  statement {
+    sid    = "AthenaQueryResultsOutput"
+    effect = "Allow"
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:ListBucketMultipartUploads",
+      "s3:ListMultipartUploadParts",
+      "s3:AbortMultipartUpload",
+      "s3:CreateBucket",
+      "s3:PutObject"
+    ]
+    resources = [
+      "${aws_s3_bucket.athena_results.arn}",
```

Remove the interpolation syntax to pass CI tests.

```suggestion
      aws_s3_bucket.athena_results.arn,
```